### PR TITLE
feat(ui): Create `IdBadge` that supports Organization, Project, Team, and User

### PR DESCRIPTION
**Pull request by billyvg**
Provides a single interface to display an "Id Badge" for the above data types.
By default only projects will not display an avatar. Also by default, will have a `max-width: 200px` for org/project/team slugs.

image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/idBadge/slugOverflow.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import styled from 'react-emotion';
+
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+
+const SlugOverflow = styled('span')`
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/idBadge/slugOverflow.jsx]: Default wrapper used when displaying team/project/org slugs. `maxSlugWidth` is currently at `200px`

File: src/sentry/static/sentry/app/utils/theme.jsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 const theme = {
-  white: '#FFFFFF',
+  /*** Colors ***/
```
Comment (Chrissy) [inline src/sentry/static/sentry/app/utils/theme.jsx]: Maybe just do `\\` here for consistency?

File: src/sentry/static/sentry/app/components/idBadge/badgeDisplayName.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import styled from 'react-emotion';
+
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import space from 'app/styles/space';
+
+const BadgeDisplayName = styled('span')`
+  ${overflowEllipsis};
+  max-width: ${p => p.theme.settings.maxCrumbWidth};
```
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/idBadge/badgeDisplayName.jsx]: This seems like a concern we don't want to worry about in every single badge.

Can you explain the problem here a little more? Maybe I can help suggest some flex-fu to move this up to `breadcrumb` somehow.

**Review by Chrissy — CHANGES_REQUESTED** [review]
Looks great! A few minor suggestions.

File: src/sentry/static/sentry/app/components/idBadge/badgeDisplayName.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/idBadge/badgeDisplayName.jsx]: So here are some examples of where this is used:
image

I was debating where this should be handled, and I think the badge makes sense (especially if we start using it everywhere we want to display a name/badge). User+other data slugs are generally the only user input that we echo back to the user, so it has the potential to disrupt layouts when not accounted for.

It'd be preferable if this wasn't a theme property and was handled by a prop though.

**Review by Chrissy — APPROVED** [review]

**Pull request merged**